File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int min_price = INT_MAX;
        
        for(int i=0; i<prices.size(); i++){
            // if(prices[i] < minI) 
            //     minI = prices[i];
            // else if(prices[i] - minI > profit)
            //     profit = prices[i] - minI;
            min_price = min(min_price, prices[i]);
            max_profit = max(max_profit, prices[i]-min_price);
        }
        return max_profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int minI = Integer.MAX_VALUE;
        
        for(int i=0; i<prices.length; i++){
            // if(prices[i] < minI) 
            //     minI = prices[i];
            // else if(prices[i] - minI > profit)
            //     profit = prices[i] - minI;
            minI = Math.min(minI, prices[i]);
            profit = Math.max(profit, prices[i] - minI);
        }
        return profit;
    }
}